File: 0001-0100/0070.climbing-stairs.cpp
// TODO: 使用矩阵乘幂
class Solution {
public:
    int climbStairs(int n) {
        if (n <= 2) return n;
        int pre = 1, cur = 1, tmp;
        for (int i = 2; i <= n; ++i) {
            tmp = cur;
            cur += pre;
            pre = tmp;
        }
        return cur;
    }
};

File: 0001-0100/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2: return n
        one = 2
        two = 1
        for i in range(3, n):
            old_one = one
            one = two + one
            two = old_one
        return one + two

print(Solution().climbStairs(2))
print(Solution().climbStairs(3))
